backend/app/synthesis/synthesizers.py

```diff
@@ -3,44 +3,103 @@
 """
 import math
 import numpy as np
-
+from scipy import signal
 
 from app.common import NOTE_FREQ_SIMPLE
 from app.synthesis.audio_encoding import WAV_SAMPLE_RATE
 
 
-def generate_sine_wave(frequency, duration):
+# pylint: disable=too-many-locals
+def generate_wave(frequency, duration, harmonics=0, vibrato=False, wave_type=np.sin):
+    """
+    Generates audio samples for various wave types
+    :param frequency: frequency in Hz
+    :param duration:  duration in seconds
+    :param harmonics: number of desired harmonics
+    :param wave_type:  wave type
+    :param vibrato: boolean for using vibrato or not
+    :return: audio samples for wave of the given wave type
+    """
+    num_samples = int(duration * WAV_SAMPLE_RATE)
+    time_steps = np.linspace(0, duration, num=num_samples, retstep=False)
+    wave_samples = wave_type(frequency * 2 * np.pi * time_steps)
+
+    if harmonics != 0:
+        for i in range(harmonics):
+            harmonic = np.sin(frequency * 2 * np.pi * time_steps * (i + 2))
+            for j, _ in enumerate(wave_samples):
+                wave_samples[j] += harmonic[j]
+
+    if vibrato:
+        vibrato_changer = 5  # 5 is arbitrary, decrease for stronger vibrato
+        copies = {}
+
+        for c in range(20):
+            copy = len(wave_samples) * [wave_samples[0] / vibrato_changer]
+            for i in range(2000 * c, len(wave_samples)):
+                if i < (len(wave_samples) - 1000):
+                    copy[i] = (wave_samples[i - 2000 * c]) / vibrato_changer
+                else:
+                    copy[i] = 0
+
+            copies[c] = copy
+
+        for i, _ in enumerate(copies):
+            wave_samples += copies[i]
+
+    return wave_samples
+
+
+def generate_sine_wave(frequency, duration, harmonics=0, vibrato=False):
```

Comment:
I'm a big fan of making these convenience functions with the various wave_type arguments specified. Will make it much easier for folks to use this module without having to know where the various wave generation functions live.